src/ray/core_worker/future_resolver.h

```diff
@@ -0,0 +1,68 @@
+#ifndef RAY_CORE_WORKER_FUTURE_RESOLVER_H
+#define RAY_CORE_WORKER_FUTURE_RESOLVER_H
+
+#include <memory>
+
+#include "ray/common/id.h"
+#include "ray/core_worker/store_provider/memory_store/memory_store.h"
+#include "ray/protobuf/core_worker.pb.h"
+#include "ray/rpc/worker/core_worker_client.h"
+
+namespace ray {
+
+/// Max time between requests to the owner to check whether the object has been
+/// evicted.
+const int kWaitObjectEvictionMilliseconds = 100;
+
+// Resolve values for futures that were given to us before the value
+// was available. This class is thread-safe.
+class FutureResolver {
+ public:
+  FutureResolver(std::shared_ptr<CoreWorkerMemoryStore> store,
```

Comment:
I think this class could be an internal component of plasma_store_provider and not exposed to core worker.

Reply:
agreed

Reply:
I think we already covered this offline, but: I don't think this really makes sense since we need to do it separately from the `Get` codepath in the plasma_store_provider.